### PR TITLE
fdisk: introduce common fdisk_new_context_fd() helper

### PR DESCRIPTION
**Pull request by poettering**
We do the same thing over and over again and it's a bit ugly, hence let's unify the code for it at one common place.

### CODE REVIEW

**Comment by poettering** [comment]
Inspired by #25453

File: src/shared/meson.build [hunk 1]
```diff
@@ -454,6 +455,7 @@ libshared_deps = [threads,
                   libcap,
                   libcrypt,
                   libdl,
+                  libfdisk,
```
Comment (bluca) [inline src/shared/meson.build]: This means we get a new core dependency, no? While before only repart/homed linked to this, is it worth it for a single helper?

Comment (poettering) [reply src/shared/meson.build]: well, i am pretty sure we'll grow more. like for example one that reads a uuid from libfdisk and parses it right away as sd_id128.

so far, we scared away for adding common helpers, but that's not sustainable. We should add common helpers, it's a good thing, not a bad thing.

Right now it's just one, but I am sure there will be more pretty soon.

**Comment by poettering** [comment]
force pushed a new version that avoids adding the dep on libshared.so

**Review by yuwata — APPROVED** [review]

File: src/partition/repart.c [hunk 2]
```diff
@@ -6033,11 +6034,7 @@ static int resize_pt(int fd) {
          * possession of the enlarged backing file. For this it suffices to open the device with libfdisk and
          * immediately write it again, with no changes. */
 
-        c = fdisk_new_context();
-        if (!c)
-                return log_oom();
-
-        r = fdisk_assign_device(c, FORMAT_PROC_FD_PATH(fd), 0);
+        r = fdisk_new_context_fd(fd, fd, /* read_only= */ false);
```
Comment (yuwata) [inline src/partition/repart.c]: The arguments are borked, and this triggers assertion.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/partition/repart.c (hunk 2 above)
Comment (poettering) [reply src/partition/repart.c]: ouch!

**Comment by poettering** [comment]
ptal. fixed the massive typo...

**Review by yuwata — APPROVED** [review]

**Pull request merged**